ngmix/tests/test_em.py

```diff
@@ -0,0 +1,326 @@
+import pytest
+import numpy as np
+
+from ngmix import DiagonalJacobian, GMix, GMixModel
+from ngmix.em import fit_em
+from ngmix import Observation
+from ngmix.priors import srandu
+from ngmix.bootstrap import PSFRunnerEM
+
+
+def get_obs(*, rng, ngauss, pixel_scale, noise=0.0, withpsf=False):
+
+    counts = 100.0
+    dims = [25, 25]
+    cen = (np.array(dims) - 1.0) / 2.0
+    jacob = DiagonalJacobian(scale=pixel_scale, row=cen[0], col=cen[1])
+
+    T_1 = 0.55  # arcsec**2
+    if withpsf:
+        Tpsf = 0.27
+        T_1 = T_1 - Tpsf
+        psf_gm = GMixModel([0.0, 0.0, 0.0, 0.0, Tpsf, 1.0], "turb")
+
+    e1_1 = 0.1
+    e2_1 = 0.05
+
+    irr_1 = T_1 / 2.0 * (1 - e1_1)
+    irc_1 = T_1 / 2.0 * e2_1
+    icc_1 = T_1 / 2.0 * (1 + e1_1)
+
+    cen1 = [-3.25*pixel_scale, -3.25*pixel_scale]
+
+    if ngauss == 2:
+
+        frac1 = 0.4
+        frac2 = 0.6
+
+        cen2 = [3.0*pixel_scale, 0.5*pixel_scale]
+
+        T_2 = T_1/2
+
+        e1_1 = 0.1
+        e2_1 = 0.05
+        e1_2 = -0.2
+        e2_2 = -0.1
+
+        counts_1 = frac1 * counts
+        counts_2 = frac2 * counts
+
+        irr_1 = T_1 / 2.0 * (1 - e1_1)
+        irc_1 = T_1 / 2.0 * e2_1
+        icc_1 = T_1 / 2.0 * (1 + e1_1)
+
+        irr_2 = T_2 / 2.0 * (1 - e1_2)
+        irc_2 = T_2 / 2.0 * e2_2
+        icc_2 = T_2 / 2.0 * (1 + e1_2)
+
+        pars = [
+            counts_1,
+            cen1[0],
+            cen1[1],
+            irr_1,
+            irc_1,
+            icc_1,
+            counts_2,
+            cen2[0],
+            cen2[1],
+            irr_2,
+            irc_2,
+            icc_2,
+        ]
+
+    elif ngauss == 1:
+
+        pars = [
+            counts,
+            cen1[0],
+            cen1[1],
+            irr_1,
+            irc_1,
+            icc_1,
+        ]
+
+    gm = GMix(pars=pars)
+
+    if withpsf:
+        gmconv = gm.convolve(psf_gm)
+        psf_im = psf_gm.make_image(dims, jacobian=jacob)
+        psf_obs = Observation(psf_im, jacobian=jacob)
+        psf_obs.set_gmix(psf_gm)
+
+        im0 = gmconv.make_image(dims, jacobian=jacob)
+    else:
+        im0 = gm.make_image(dims, jacobian=jacob)
+        psf_obs = None
+
+    im = im0 + rng.normal(size=im0.shape, scale=noise)
+    obs = Observation(im, jacobian=jacob, psf=psf_obs)
+
+    return obs, gm
+
+
+def get_psf_obs():
+    pixel_scale = 0.263
+    dims = [25, 25]
+    cen = (np.array(dims) - 1.0) / 2.0
+
+    jacob = DiagonalJacobian(scale=pixel_scale, row=cen[0], col=cen[1])
+
+    Tpsf = 0.27
+    psf_gm = GMixModel([0.0, 0.0, 0.0, 0.0, Tpsf, 1.0], "turb")
+    # psf_gm = GMixModel([0.0, 0.0, 0.0, 0.0, Tpsf, 1.0], "gauss")
+    psf_im = psf_gm.make_image(dims, jacobian=jacob)
+
+    return Observation(psf_im, jacobian=jacob), psf_gm
+
+
+@pytest.mark.parametrize('noise', [0.0, 0.05])
+def test_1gauss(noise):
```

Comment:
```suggestion
def test_em_1gauss(noise):
```

I've been trying to use this convention so it is easy to link the unit test functions back to the modules they target.